indra/llcharacter/llik.cpp

```diff
@@ -271,35 +275,26 @@ size_t LLIK::Constraint::generateHash() const
     return seed;
 }
 
-
+// returns 'true' if joint was adjusted
 bool LLIK::Constraint::enforce(Joint& joint) const
 {
+    // This enforce() method of the base-Constraint provides no back-pressure
+    // on joint's parent when adjusting joint's local-frame transform.
+    // Override this method as necessary to provide back-pressure for select
+    // derived Constraints.
     const LLQuaternion& local_rot = joint.getLocalRot();
     LLQuaternion adjusted_local_rot = computeAdjustedLocalRot(local_rot);
     if (! LLQuaternion::almost_equal(adjusted_local_rot, local_rot))
     {
+        // Note: we update joint's local-frame rot, but not its world-frame rot
+        // -- that responsibility belongs to external code.
         joint.setLocalRot(adjusted_local_rot);
+        joint.applyLocalRot();
         return true;
     }
     return false;
 }
 
-LLQuaternion LLIK::Constraint::minimizeTwist(const LLQuaternion& joint_local_rot) const
-{
-    // Default behavior of minimizeTwist() is to compute
-    // the shortest rotation that produces the same swing.
-    LLVector3 joint_forward = mForward * joint_local_rot;
-    LLVector3 swing_axis = mForward % joint_forward;
-    constexpr F32 MIN_AXIS_LENGTH = 1.0e-5f;
-    LLQuaternion new_local_rot = LLQuaternion::DEFAULT;
-    if (swing_axis.length() > MIN_AXIS_LENGTH)
-    {
-        F32 swing_angle = std::acos(mForward * joint_forward);
-        new_local_rot.setAngleAxis(swing_angle, swing_axis);
-    }
-    return new_local_rot;
-}
-
```

Comment:
I'm removing the `minimizeTwist()` methods.  I've learned: it is better to perform twist adjustments inside the `Constraint::enforce()` overrides.  See further below.